deploy/lib/application-stack.ts

```diff
@@ -51,50 +54,69 @@ export class ApplicationStack extends Stack {
       blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
     });
 
+    if (props.destinationAccountId) {
+      clientBucket.grantRead(new AccountPrincipal(props.destinationAccountId));
+    }
+
     const distribution = this.setupS3CloudFrontIntegration(clientBucket, props.aliasDomainName);
-    this.buildReactApp(clientBucket, props.reactBuildEnvVariables, distribution);
+    this.buildReactApp(
+      clientBucket,
+      props.reactBuildEnvVariables,
+      distribution,
+      props.sourceBucketName
+    );
   }
 
   private buildReactApp(
     bucket: IBucket,
     reactBuildEnvVariables: Record<string, string>,
-    distribution: Distribution
+    distribution: Distribution,
+    sourceBucketName: string
   ) {
-    const codeStarArn = StringParameter.valueForStringParameter(this, 'codestar_github_arn');
-
-    const sourceOutput = new Artifact();
-    const buildOutput = new Artifact();
+    const sourceBucket = Bucket.fromBucketName(this, 'SourceBucket', sourceBucketName);
 
-    const buildProject = new PipelineProject(this, 'ReactBuildCodeBuildProject', {
+    const project = new Project(this, 'ReactBuildCodeBuildProject', {
       buildSpec: BuildSpec.fromObject({
         version: 0.2,
         phases: {
           install: {
             'runtime-versions': {
               nodejs: 20,
             },
-            commands: ['node -v', 'corepack enable', 'yarn --version', 'yarn install --immutable'],
+            commands: [
+              'aws s3 rm s3://${VITE_BUCKET_NAME}/ --recursive',
+              'aws s3 sync ./ s3://${VITE_BUCKET_NAME}/${SOURCE_BUCKET_ARTIFACT_PATH} --dryrun',
```

Comment:
`--dryrun` can be removed.

For 'chains assets between the stage deployment', it will keep what we want like, just deploy what we have in the last stage. 

I am thinking if we just use the deployment in bastion, we still need to face the problem, which version of code we need retrieve? as we alway just retrieve the code from the "latest context" and build the original atrtifact for each stage. I am not find a proper way to have commit SHA in the checking out codestar source code, only from github link or use lambda. I will keep checking...
So I am think kind of store each stage artifact in each stage bucket and keep the version we have in the previous stage...

Reply:
If the build is in the bastion account and deployed to its respective (cloudfront) bucket, it could use the same source artifact for each stage deployment within the code pipeline trigger (commit). I suppose code pipeline will keep its artifact version for each trigger.

Reply:
yes, if we are in one pipeline, sure, because there will always be a new build if new commit come in to trigger the build. but the bucket not. So I am think use that artifacts bucket
Later, I will refactor it again to fit it in the one pipeline in the bastion account.

Reply:
> if we are in one pipeline,
We are only using one pipeline right?

> but the bucket not.
What do you mean the bucket is not? What I had in mind is for every stage the build should source from the artifact (github commit) that trigger the pipeline, build the React and send it off to its respective cf (deployment) bucket.